src/v/cluster/types.h

```diff
@@ -360,24 +362,31 @@ struct incremental_topic_updates {
     property_update<tristate<size_t>> retention_bytes;
     property_update<tristate<std::chrono::milliseconds>> retention_duration;
 
-    // Data-policy property is replicated by data_policy_frontend and handled by
-    // data_policy_manager.
-    property_update<std::optional<v8_engine::data_policy>> data_policy;
-
     friend bool operator==(
       const incremental_topic_updates&, const incremental_topic_updates&)
       = default;
 };
 
+// This class contains updates for topic properties which are replicates not by
+// topic_frontend
+struct incremental_topic_custom_updates {
+    // Data-policy property is replicated by data_policy_frontend and handled by
+    // data_policy_manager.
+    property_update<std::optional<v8_engine::data_policy>> data_policy;
+};
+
 /**
  * Struct representing single topic properties update
  */
 struct topic_properties_update {
+    // We need version to indetify request with custom_properties
+    static constexpr int32_t version = -1;
```

Comment:
Problem with version here:
1) old redpanda(replica) send topic_properties_update to new redpanda version(leader)
2) It will parse  incremental_topic_updates which has version -1 and skip redpanda
3) new version redpanda can not get datapolicy property from request

I can parse this rpc rrequest by hand, and parse old version incremental_topic_updates not with adl<incremental_topic_updates> stuff. parse each field and save data_policy

Do we need?

Reply:
If this was something we were already supporting customers on then we would definitely need that, but I think coproc is not at that point yet, so it's probably fine for a mid-upgrade cluster to drop the data policy on the floor in the even that this scenario comes up.